api/v3/Activity.php

```diff
@@ -63,71 +63,14 @@ function civicrm_api3_activity_create($params) {
-    $params['case_id'] = CRM_Core_DAO::singleValueQuery("SELECT case_id FROM civicrm_case_activity WHERE activity_id = " . (int) $params['id']);
-  }
-  if (!empty($params['case_id'])) {
-    $case_id = $params['case_id'];
-    if (!empty($params['id']) && Civi::settings()->get('civicaseActivityRevisions')) {
-      $oldActivityParams = ['id' => $params['id']];
-      if (!$oldActivityValues) {
-        CRM_Activity_BAO_Activity::retrieve($oldActivityParams, $oldActivityValues);
-      }
-      if (empty($oldActivityValues)) {
-        throw new API_Exception(ts("Unable to locate existing activity."));
-      }
-      else {
-        $activityDAO = new CRM_Activity_DAO_Activity();
-        $activityDAO->id = $params['id'];
-        $activityDAO->is_current_revision = 0;
-        if (!$activityDAO->save()) {
-          throw new API_Exception(ts("Unable to revision existing case activity."));
-        }
-        $createRevision = TRUE;
-      }
-    }
-  }
-
-  if ($case_id && $createRevision) {
-    // This is very similar to the copy-to-case action.
-    if (!CRM_Utils_Array::crmIsEmptyArray($oldActivityValues['target_contact'])) {
-      $oldActivityValues['targetContactIds'] = implode(',', array_unique($oldActivityValues['target_contact']));
-    }
-    if (!CRM_Utils_Array::crmIsEmptyArray($oldActivityValues['assignee_contact'])) {
-      $oldActivityValues['assigneeContactIds'] = implode(',', array_unique($oldActivityValues['assignee_contact']));
-    }
-    $oldActivityValues['mode'] = 'copy';
-    $oldActivityValues['caseID'] = $case_id;
-    $oldActivityValues['activityID'] = $oldActivityValues['id'];
-    $oldActivityValues['contactID'] = $oldActivityValues['source_contact_id'];
-
-    $copyToCase = CRM_Activity_Page_AJAX::_convertToCaseActivity($oldActivityValues);
-    if (empty($copyToCase['error_msg'])) {
-      // now fix some things that are different from copy-to-case
-      // then fall through to the create below to update with the passed in params
-      $params['id'] = $copyToCase['newId'];
-      $params['is_auto'] = 0;
-      $params['original_id'] = empty($oldActivityValues['original_id']) ? $oldActivityValues['id'] : $oldActivityValues['original_id'];
-    }
-    else {
-      throw new API_Exception(ts("Unable to create new revision of case activity."));
-    }
-  }
-
   // create activity
   $activityBAO = CRM_Activity_BAO_Activity::create($params);
 
   if (isset($activityBAO->id)) {
-    if ($case_id && $isNew && !$createRevision) {
+    // Fixme - Move business logic out of API
+    if (!empty($params['case_id']) && $isNew) {
       // If this is a brand new case activity, add to case(s)
-      foreach ((array) $case_id as $singleCaseId) {
+      foreach ((array) $params['case_id'] as $singleCaseId) {
         $caseActivityParams = ['activity_id' => $activityBAO->id, 'case_id' => $singleCaseId];
         CRM_Case_BAO_Case::processCaseActivity($caseActivityParams);
```

Comment:
This right here is the motivation for this PR. I need to get `$params['case_id']` working in APIv4 `Activity.create` and the best solution is to move this logic to the BAO.
See #24244